skxray/core/speckle.py

```diff
@@ -0,0 +1,290 @@
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT    #
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS    #
+# FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE       #
+# COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,           #
+# INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES   #
+# (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR   #
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)   #
+# HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,  #
+# STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OTHERWISE) ARISING   #
+# IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE   #
+# POSSIBILITY OF SUCH DAMAGE.                                          #
+########################################################################
+
+"""
+    X-ray speckle visibility spectroscopy(XSVS) - Dynamic information of
+    the speckle patterns  are obtained by analyzing the photon statistics
+    and calculating the speckle contrast in single scattering patterns.
+
+    This module will provide XSVS analysis tools
+"""
+
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+import six
+import numpy as np
+import time
+
+import skxray.core.correlation as corr
+import skxray.core.roi as roi
+from skxray.core.utils import bin_edges_to_centers, geometric_series
+
+import logging
+logger = logging.getLogger(__name__)
+
+
+def xsvs(image_sets, label_array, number_of_img, timebin_num=2,
+         max_cts=None):
+    """
+    This function will provide the probability density of detecting speckles
+    for different integration times.
+
+    The experimental probability density P(K) of detecting speckles K is
+    obtained by histogramming the photon counts over an ensemble of
+    equivalent pixels and over a number of speckle patterns recorded
+    with the same integration time T under the same condition.
+
+    Parameters
+    ----------
+    image_sets : array
+        sets of images
+    label_array : array
+        labeled array; 0 is background.
+        Each ROI is represented by a distinct label (i.e., integer).
+    number_of_img : int
+        number of images (how far to go with integration times when finding
+        the time_bin, using skxray.utils.geometric function)
+    timebin_num : int, optional
+        integration time; default is 2
+    max_cts : int, optional
+       the brightest pixel in any ROI in any image in the image set.
+       default None; if not provided can be use roi_max_counts function
+       in skxray.roi module
+
+    Returns
+    -------
+    prob_k_all : array
+        probability density of detecting speckles
+    prob_k_std_dev : array
+        standard deviation of probability density of detecting speckles
+
+    Notes
+    -----
+    These implementation is based on following references
```

Comment:
Can you put in a PR with your speckle notebook against scikit-xray-examples and add another bullet to this section noting that there is an example in https://github.com/scikit-xray/scikit-xray-examples with that notebook?